librosa/onset.py

```diff
@@ -602,22 +602,27 @@ def onset_strength_multi(
         pad = False
 
     if aggregate:
-        onset_env = util.sync(onset_env, channels, aggregate=aggregate, pad=pad, axis=0)
+        onset_env = util.sync(onset_env, channels, aggregate=aggregate, pad=pad, axis=-2)
 
     # compensate for lag
     pad_width = lag
     if center:
         # Counter-act framing effects. Shift the onsets by n_fft / hop_length
         pad_width += n_fft // (2 * hop_length)
 
-    onset_env = np.pad(onset_env, ([0, 0], [int(pad_width), 0]), mode="constant")
+    onset_env_ndims = onset_env.ndim
+    onset_env = np.pad(onset_env, 
+                    [[0,0] if i != (onset_env_ndims-1) else [int(pad_width),0] 
+                        for i in range(onset_env_ndims)], 
```

Comment:
I'd prefer to expand this list comprehension out into something more readable:
```python
padding = [(0,0) for _ in onset_env.shape]
padding[-1] = (int(pad_width), 0)
onset_env = np.pad(onset_env, padding, mode='constant')
```

We use this pattern elsewhere, and it's a bit more readable IMO.